### ISSUE TITLE
Cannot create repository,vs crash

### ISSUE THREAD

**Issue by TracyDa** [github/VisualStudio#503]
I cannot use it to create repository ,it will make my visual studio crash.
Really disappointed！

**Comment by shana** [comment]
Can you post screenshots of the steps you do up to the crash, and send us the logs as described in https://github.com/github/VisualStudio/blob/master/CONTRIBUTING.md#submitting-an-issue? Thanks!

**Comment by TracyDa** [comment]
OK,I have sent you email with the screenshot and log file.Hope you can fix it.

**Comment by shana** [comment]
@TracyDa Thanks, this is helpful!

Looks like we have a bug handling the dropdown list in certain locales:
```
System.NullReferenceException: δ??????????????ʵ????
   ? GitHub.UI.IsLastItemInContainerConverter.Convert(Object value, Type targetType, Object parameter, CultureInfo culture) λ? Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\src\GitHub.UI\Converters\IsLastItemInContainerConverter.cs:???17
   ? System.Windows.Data.BindingExpression.TransferValue(Object newValue, Boolean isASubPropertyChange)
   ? System.Windows.Data.BindingExpression.HandlePropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   ? System.Windows.Data.BindingExpressionBase.OnPropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   ? System.Windows.Data.BindingExpression.OnPropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   ? System.Windows.DependentList.InvalidateDependents(DependencyObject source, DependencyPropertyChangedEventArgs sourceArgs)
   ? System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   ? System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   ? System.Windows.TreeWalkHelper.InvalidateTreeDependentProperty(TreeChangeInfo info, DependencyObject d, FrameworkObject& fo, DependencyProperty dp, FrameworkPropertyMetadata fMetadata, Style selfStyle, Style selfThemeStyle, ChildRecord& childRecord, Boolean isChildRecordValid, Boolean hasStyleChanged, Boolean isSelfInheritanceParent)
   ? System.Windows.TreeWalkHelper.InvalidateTreeDependentProperties(TreeChangeInfo info, FrameworkElement fe, FrameworkContentElement fce, Style selfStyle, Style selfThemeStyle, ChildRecord& childRecord, Boolean isChildRecordValid, Boolean hasStyleChanged, Boolean isSelfInheritanceParent)
   ? System.Windows.FrameworkElement.InvalidateTreeDependentProperties(TreeChangeInfo parentTreeState, Boolean isSelfInheritanceParent)
   ? System.Windows.FrameworkElement.OnAncestorChangedInternal(TreeChangeInfo parentTreeState)
   ? System.Windows.TreeWalkHelper.OnAncestorChanged(DependencyObject d, TreeChangeInfo info, Boolean visitedViaVisualTree)
   ? System.Windows.DescendentsWalker`1.StartWalk(DependencyObject startNode, Boolean skipStartNode)
   ? MS.Internal.PrePostDescendentsWalker`1.StartWalk(DependencyObject startNode, Boolean skipStartNode)
   ? System.Windows.TreeWalkHelper.InvalidateOnTreeChange(FrameworkElement fe, FrameworkContentElement fce, DependencyObject parent, Boolean isAddOperation)
   ? System.Windows.FrameworkElement.OnVisualParentChanged(DependencyObject oldParent)
   ? System.Windows.Controls.ListBoxItem.OnVisualParentChanged(DependencyObject oldParent)
   ? System.Windows.Media.Visual.FireOnVisualParentChanged(DependencyObject oldParent)
   ? System.Windows.Media.Visual.RemoveVisualChild(Visual child)
   ? System.Windows.Media.VisualCollection.DisconnectChild(Int32 index)
   ? System.Windows.Media.VisualCollection.Clear()
   ? System.Windows.Controls.UIElementCollection.ClearInternal()
   ? System.Windows.Controls.Panel.ClearChildren()
   ? System.Windows.Controls.Panel.EnsureEmptyChildren(FrameworkElement logicalParent)
   ? System.Windows.Controls.Panel.OnItemsChangedInternal(Object sender, ItemsChangedEventArgs args)
   ? System.Windows.Controls.Panel.OnItemsChanged(Object sender, ItemsChangedEventArgs args)
   ? System.Windows.Controls.ItemContainerGenerator.OnRefresh()
   ? System.Windows.Controls.ItemContainerGenerator.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)
   ? System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)
   ? System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)
   ? System.Collections.Specialized.CollectionChangedEventManager.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)
   ? System.Collections.Specialized.NotifyCollectionChangedEventHandler.Invoke(Object sender, NotifyCollectionChangedEventArgs e)
   ? System.Windows.Data.CollectionView.OnCollectionChanged(NotifyCollectionChangedEventArgs args)
   ? System.Windows.Controls.ItemCollection.OnViewCollectionChanged(Object sender, NotifyCollectionChangedEventArgs e)
   ? System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)
   ? System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)
   ? System.Collections.Specialized.CollectionChangedEventManager.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)
   ? System.Windows.Data.CollectionView.OnCollectionChanged(NotifyCollectionChangedEventArgs args)
   ? System.Windows.Data.ListCollectionView.RefreshOverride()
   ? System.Windows.Data.CollectionView.RefreshInternal()
   ? System.Windows.Data.CollectionView.RefreshOrDefer()
   ? System.Windows.Data.CollectionView.set_Filter(Predicate`1 value)
   ? System.Windows.Data.ListCollectionView.set_Filter(Predicate`1 value)
   ? System.Windows.Controls.ItemCollection.set_Filter(Predicate`1 value)
   ? GitHub.UI.FilteredComboBox.<.ctor>b__1_1(String filterText) λ? Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\src\GitHub.UI.Reactive\Controls\FilteredComboBox.cs:???26
   ? System.Reactive.AnonymousSafeObserver`1.OnNext(T value)
   ? System.Reactive.Linq.ObservableImpl.Throttle`1._.Propagate(IScheduler self, UInt64 currentid)
   ? System.Reactive.Concurrency.DispatcherScheduler.<>c__DisplayClass13_0`1.<Schedule>b__0(Object s, EventArgs e)
   ? System.Windows.Threading.DispatcherTimer.FireTick(Object unused)
   ? System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   ? System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   ? System.Windows.Threading.DispatcherOperation.InvokeImpl()
   ? System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   ? System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   ? System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   ? System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   ? MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   ? System.Windows.Threading.DispatcherOperation.Invoke()
   ? System.Windows.Threading.Dispatcher.ProcessQueue()
   ? System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   ? MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   ? MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   ? System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   ? System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   ? System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   ? MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   ? MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   ? System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   ? System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   ? System.Windows.Window.ShowHelper(Object booleanBox)
   ? System.Windows.Window.Show()
   ? System.Windows.Window.ShowDialog()
   ? Microsoft.Internal.VisualStudio.PlatformUI.WindowHelper.ShowModal(Window window, IntPtr parent)
   ? Microsoft.VisualStudio.PlatformUI.DialogWindow.ShowModal()
   ? GitHub.VisualStudio.UIProvider.RunUI() λ? Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\src\GitHub.VisualStudio\Services\UIProvider.cs:???279    ? GitHub.UI.IsLastItemInContainerConverter.Convert(Object value, Type targetType, Object parameter, CultureInfo culture) λ? Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\src\GitHub.UI\Converters\IsLastItemInContainerConverter.cs:???17
   ? System.Windows.Data.BindingExpression.TransferValue(Object newValue, Boolean isASubPropertyChange)
   ? System.Windows.Data.BindingExpression.HandlePropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   ? System.Windows.Data.BindingExpressionBase.OnPropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   ? System.Windows.Data.BindingExpression.OnPropertyInvalidation(DependencyObject d, DependencyPropertyChangedEventArgs args)
   ? System.Windows.DependentList.InvalidateDependents(DependencyObject source, DependencyPropertyChangedEventArgs sourceArgs)
   ? System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   ? System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   ? System.Windows.TreeWalkHelper.InvalidateTreeDependentProperty(TreeChangeInfo info, DependencyObject d, FrameworkObject& fo, DependencyProperty dp, FrameworkPropertyMetadata fMetadata, Style selfStyle, Style selfThemeStyle, ChildRecord& childRecord, Boolean isChildRecordValid, Boolean hasStyleChanged, Boolean isSelfInheritanceParent)
   ? System.Windows.TreeWalkHelper.InvalidateTreeDependentProperties(TreeChangeInfo info, FrameworkElement fe, FrameworkContentElement fce, Style selfStyle, Style selfThemeStyle, ChildRecord& childRecord, Boolean isChildRecordValid, Boolean hasStyleChanged, Boolean isSelfInheritanceParent)
   ? System.Windows.FrameworkElement.InvalidateTreeDependentProperties(TreeChangeInfo parentTreeState, Boolean isSelfInheritanceParent)
   ? System.Windows.FrameworkElement.OnAncestorChangedInternal(TreeChangeInfo parentTreeState)
   ? System.Windows.TreeWalkHelper.OnAncestorChanged(DependencyObject d, TreeChangeInfo info, Boolean visitedViaVisualTree)
   ? System.Windows.DescendentsWalker`1.StartWalk(DependencyObject startNode, Boolean skipStartNode)
   ? MS.Internal.PrePostDescendentsWalker`1.StartWalk(DependencyObject startNode, Boolean skipStartNode)
   ? System.Windows.TreeWalkHelper.InvalidateOnTreeChange(FrameworkElement fe, FrameworkContentElement fce, DependencyObject parent, Boolean isAddOperation)
   ? System.Windows.FrameworkElement.OnVisualParentChanged(DependencyObject oldParent)
   ? System.Windows.Controls.ListBoxItem.OnVisualParentChanged(DependencyObject oldParent)
   ? System.Windows.Media.Visual.FireOnVisualParentChanged(DependencyObject oldParent)
   ? System.Windows.Media.Visual.RemoveVisualChild(Visual child)
   ? System.Windows.Media.VisualCollection.DisconnectChild(Int32 index)
   ? System.Windows.Media.VisualCollection.Clear()
   ? System.Windows.Controls.UIElementCollection.ClearInternal()
   ? System.Windows.Controls.Panel.ClearChildren()
   ? System.Windows.Controls.Panel.EnsureEmptyChildren(FrameworkElement logicalParent)
   ? System.Windows.Controls.Panel.OnItemsChangedInternal(Object sender, ItemsChangedEventArgs args)
   ? System.Windows.Controls.Panel.OnItemsChanged(Object sender, ItemsChangedEventArgs args)
   ? System.Windows.Controls.ItemContainerGenerator.OnRefresh()
   ? System.Windows.Controls.ItemContainerGenerator.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)
   ? System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)
   ? System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)
   ? System.Collections.Specialized.CollectionChangedEventManager.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)
   ? System.Collections.Specialized.NotifyCollectionChangedEventHandler.Invoke(Object sender, NotifyCollectionChangedEventArgs e)
   ? System.Windows.Data.CollectionView.OnCollectionChanged(NotifyCollectionChangedEventArgs args)
   ? System.Windows.Controls.ItemCollection.OnViewCollectionChanged(Object sender, NotifyCollectionChangedEventArgs e)
   ? System.Windows.WeakEventManager.ListenerList`1.DeliverEvent(Object sender, EventArgs e, Type managerType)
   ? System.Windows.WeakEventManager.DeliverEvent(Object sender, EventArgs args)
   ? System.Collections.Specialized.CollectionChangedEventManager.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs args)
   ? System.Windows.Data.CollectionView.OnCollectionChanged(NotifyCollectionChangedEventArgs args)
   ? System.Windows.Data.ListCollectionView.RefreshOverride()
   ? System.Windows.Data.CollectionView.RefreshInternal()
   ? System.Windows.Data.CollectionView.RefreshOrDefer()
   ? System.Windows.Data.CollectionView.set_Filter(Predicate`1 value)
   ? System.Windows.Data.ListCollectionView.set_Filter(Predicate`1 value)
   ? System.Windows.Controls.ItemCollection.set_Filter(Predicate`1 value)
   ? GitHub.UI.FilteredComboBox.<.ctor>b__1_1(String filterText) λ? Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\src\GitHub.UI.Reactive\Controls\FilteredComboBox.cs:???26
   ? System.Reactive.AnonymousSafeObserver`1.OnNext(T value)
   ? System.Reactive.Linq.ObservableImpl.Throttle`1._.Propagate(IScheduler self, UInt64 currentid)
   ? System.Reactive.Concurrency.DispatcherScheduler.<>c__DisplayClass13_0`1.<Schedule>b__0(Object s, EventArgs e)
   ? System.Windows.Threading.DispatcherTimer.FireTick(Object unused)
   ? System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   ? System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   ? System.Windows.Threading.DispatcherOperation.InvokeImpl()
   ? System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   ? System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   ? System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   ? System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   ? MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   ? System.Windows.Threading.DispatcherOperation.Invoke()
   ? System.Windows.Threading.Dispatcher.ProcessQueue()
   ? System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   ? MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   ? MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   ? System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   ? System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   ? System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   ? MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   ? MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   ? System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   ? System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   ? System.Windows.Window.ShowHelper(Object booleanBox)
   ? System.Windows.Window.Show()
   ? System.Windows.Window.ShowDialog()
   ? Microsoft.Internal.VisualStudio.PlatformUI.WindowHelper.ShowModal(Window window, IntPtr parent)
   ? Microsoft.VisualStudio.PlatformUI.DialogWindow.ShowModal()
   ? GitHub.VisualStudio.UIProvider.RunUI() λ? Y:\jenkins\workspace\GitHub for Visual Studio speakeasy jenkinskeep\src\GitHub.VisualStudio\Services\UIProvider.cs:???279
```

**Comment by grokys** [comment]
I've looked into this but that method:

https://github.com/github/VisualStudio/blob/master/src/GitHub.UI/Converters/IsLastItemInContainerConverter.cs#L15

Doesn't look like it's doing anything that would be related to locale. I think the only thing that can be happening is that somehow:

    ItemsControl.ItemsControlFromItemContainer(item)

Is returning `null`. As this is happening in `FilteredComboxBox`, I'm wondering thinking the problem may be that the combo box item has been excluded from the owning `ItemsControl`, but its `ControlTemplate` still has a binding which uses `IsLastItemInContainerConverter`.

I've not been able to repro, but if that is the case I think a simple null check would suffice.

**Comment by shana** [comment]
@TracyDa We're working on a fix for this, which will be available on our next release coming out in a few days.

**Comment by TracyDa** [comment]
@shana  Thanks.

**Comment by shana** [comment]
The fix for this is out in [release 2.0.14.4](https://github.com/github/VisualStudio/releases/tag/v2.0.14.4), available from the Visual Studio gallery or from the website!
